lgc/patch/PatchWorkarounds.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2019-2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  PatchWorkarounds.cpp
```

Comment:
I suggest we rename it to PatchImageDescWorkaround currently since most codes serve this. If we insert more workarounds in this pass to make it more general. The framework in runOnModule might become such:

```
runOnModule {
  // Patch ..
  patchXxxx();

  // Patch ..
  patchYyyy();

  // Patch ..
}
```

And each workaround is checked and is applied in their own patchXxx() method.

Reply:
You mean the filename/pass?
Then change it if more workarounds are added to the runOnModule?

Reply:
Or do you mean the processImageIntrinsic (and remove the code from runOnModule - replace with a single call).

Reply:
Yes, what I mean is the same as you said. Current pass name and file are very general, but our codes are all relevant to a particular workaround. So I suggest either 1 or 2:
1. Make the pass name or file name changed to match the workaround. Change the code structure if more workarounds are added.
2. Or add a function to do the patching work and move all/most codes within it. Thus, we can easily add more workarounds. Each workaround could have such naming "applyXxxWorkaround()".

Reply:
Ok - I'm currently implementing option 2 - will post shortly...